### PR TITLE
feat: layout component type docs

### PR DESCRIPTION
**Pull request by SiTaggart**
Description and useful links to discussions / issues / tickets

### Change One

Details of change...

### Change Two

Details of change...

## Checklist

- [ ] Work in a "Draft" branch whilst you are getting feedback to reduce Chromatic costs. Once you are ready for approval, convert the PR to "Ready to review"
- [ ] Ensure all `required` checks have passed
- [ ] Icon changes must have a product design review
- [ ] Website content changes must have a product design review
- [ ] At least two engineers have reviewed any code changes
- [ ] At least one engineering lead has reviewed any core package changes or repository infrastructure
- [ ] Website visual regression testing is run by adding `🕵🏻‍♀️ Run website visual regression` label

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [ ] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

**Review by TheSisb — APPROVED** [review]

**Review by nkrantz — APPROVED** [review]

File: packages/paste-core/layout/aspect-ratio/type-docs.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "AspectRatio": {
+    "ratio": {
+      "type": "string",
+      "defaultValue": null,
+      "required": true,
+      "externalProp": false,
+      "description": "Determines the aspect ratio of the element. Use a colon separated number pattern (width:height)."
```
Comment (serifluous) [inline packages/paste-core/layout/aspect-ratio/type-docs.json:8]:
```suggestion
      "description": "Determines the aspect ratio of the element. Use a colon-separated number pattern (width:height)."
```

File: packages/paste-core/layout/grid/type-docs.json [hunk 2]
```diff
@@ -0,0 +1,3461 @@
...
+      "defaultValue": null,
+      "required": false,
+      "externalProp": true
+    },
+    "span": {
+      "type": "| number\n  | (number | null)[]\n  | { [x: string]: number | undefined; [x: number]: number | undefined }",
+      "defaultValue": null,
+      "required": false,
+      "externalProp": false,
+      "description": "Sets the width of Columns (how many columns a column can span)"
```
Comment (serifluous) [inline packages/paste-core/layout/grid/type-docs.json:3380]:
```suggestion
      "description": "Sets the width of a column by defining the number of columns that column can span"
```

File: packages/paste-website/src/pages/components/stack/index.mdx
Comment (serifluous) [inline packages/paste-website/src/pages/components/stack/index.mdx:1]: chore: Can you update the 2 "spacing tokens" links on this page to `/tokens/list#spacings`?

File: packages/paste-core/layout/aspect-ratio/src/AspectRatio.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+const handlePropValidation = ({ratio}: AspectRatioProps): void => {
+  const hasRatio = ratio != null && ratio !== '';
+
+  if (!hasRatio) {
+    throw new Error(`[Paste: AspectRatio] Missing 'ratio' prop.`);
+  }
+
+  if (!isCorrectPattern(ratio)) {
+    throw new Error(`[Paste: AspectRatio] 'ratio' is invalid. Use a colon separated number pattern (4:3).`);
```
Comment (serifluous) [inline packages/paste-core/layout/aspect-ratio/src/AspectRatio.tsx:28]:
```suggestion
    throw new Error(`[Paste: AspectRatio] 'ratio' is invalid. Use a colon-separated number pattern (4:3).`);
```

File: packages/paste-core/layout/grid/src/types.ts [hunk 4]
```diff
@@ -33,12 +59,32 @@ export interface ColumnStyleProps extends Omit<LayoutProps, 'minWidth' | 'width'
...
   gutter?: Space;
+  /**
+   * Sets the left margin of a Column (how many columns a column is offset)
+   *
+   * @type {ColumnOffset}
+   * @memberof ColumnProps
+   */
   offset?: ColumnOffset;
+  /**
+   * Sets the width of Columns (how many columns a column can span)
```
Comment (serifluous) [inline packages/paste-core/layout/grid/src/types.ts:83]:
```suggestion
   * Sets the width of a column by defining the number of columns that column can span
```

**Review by nkrantz — APPROVED** [review]

**Review by nkrantz — DISMISSED** [review]

**Review by serifluous — APPROVED** [review]

**Pull request merged**